codalab/lib/path_util.py

```diff
@@ -298,6 +301,75 @@ def rename(old_path, new_path):
     subprocess.call(['mv', old_path, new_path])
 
 
+class StorageType(Enum):
+    FILE_STORAGE = 0
+    AZURE_BLOB_STORAGE = 1
+
+
+@dataclass(frozen=True)
+class LinkedBundlePath:
+    """A LinkedBundlePath refers to a path that points to the location of a linked bundle within a specific storage location.
+    It can either point directly to the bundle, or to a file that is located within that bundle.
+    It is constructed by parsing a given bundle link URL by calling parse_bundle_url().
+
+    Attributes:
+        storage_type (StorageType): Which storage type is used to store this bundle.
+
+        bundle_path (str): Path to the bundle contents in that particular storage.
+
+        is_zip (bool): Whether this bundle is stored as a zip file on this storage medium stores folders. Only done currently by Azure Blob Storage.
+
+        uses_beam (bool): Whether this bundle's storage type requires using Apache Beam to interact with it.
+        
+        zip_subpath (str): If is_zip is True, returns the subpath within the zip file for the file that this BundlePath points to.
+
+        bundle_uuid (str): UUID of the bundle that this path refers to.
+    """
+
+    storage_type: StorageType
+    bundle_path: str
+    is_zip: bool
+    uses_beam: bool
+    zip_subpath: str
+    bundle_uuid: str
+
+
+def parse_linked_bundle_url(url):
```

Comment:
Would it be cleaner if we just made this logic be part of the constructor of `LinkedBundlePath` that would take in the url?

Reply:
I think it's a bit complicated to override constructors for frozen dataclasses; can't think of a nice way to make this work that way.